services/filemanager/s3manager.go

```diff
@@ -193,7 +197,12 @@ func (manager *S3Manager) getSession(ctx context.Context) (*session.Session, err
 		manager.Config.Region = aws.String(region)
 	}
 
-	return awsutils.CreateSession(manager.getSessionConfig())
+	var err error
+	manager.session, err = awsutils.CreateSession(manager.getSessionConfig())
```

Comment:
As it turns out, previously we were not reusing the session, but creating a new one every time. How do we verify now that reusing the session is safe and won't cause any regression?

Reply:
// This token provider will be called when ever the assumed role's
    // credentials need to be refreshed. Within the context of service clients
    // all sharing the same session the SDK will ensure calls to the token
    // provider are atomic. When sharing a token provider across multiple
    // sessions additional synchronization logic is needed to ensure the
    // token providers do not introduce race conditions. It is recommend to
    // share the session where possible.
https://docs.aws.amazon.com/sdk-for-go/api/aws/session/

Reply:
Please explain how is this documentation excerpt relevant? My understanding is that it has to do with MFA. How is it possible for us to be using MFA without third-party (i.e. user input)?

Reply:
Sorry my bad, here is right excerpt:
Sessions are safe to use concurrently as long as the Session is not being modified. Sessions should be cached when possible, because creating a new Session will load all configuration values from the environment, and config files each time the Session is created. Sharing the Session value across all of your service clients will ensure the configuration is loaded the fewest number of times possible.
https://docs.aws.amazon.com/sdk-for-go/api/aws/session/

Reply:
@atzoum can we resolve this?

Reply:
Fine, let's have our eyes open though for any possible regression due to this session reuse, hopefully there won't be any @rudderlabs/server_team

Reply:
Sure